sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/webapi/AADOAuth2OboAuthorizedClientRepository.java

```diff
@@ -0,0 +1,148 @@
+                azureClientRegistrationRepository.findByRegistrationId(registrationId);
+
+            String accessToken = authenticationToken.getToken().getTokenValue();
+            OnBehalfOfParameters parameters = OnBehalfOfParameters
+                .builder(clientRegistration.getScopes(), new UserAssertion(accessToken))
+                .build();
+            ConfidentialClientApplication clientApplication =
+                getClientApplication(clientRegistration.getRegistrationId());
+            String oboAccessToken = clientApplication.acquireToken(parameters).get().accessToken();
+
+            JWT parser = JWTParser.parse(oboAccessToken);
+            Date iat = (Date) parser.getJWTClaimsSet().getClaim("iat");
+            Date exp = (Date) parser.getJWTClaimsSet().getClaim("exp");
+            OAuth2AccessToken oAuth2AccessToken = new OAuth2AccessToken(OAuth2AccessToken.TokenType.BEARER,
+                oboAccessToken,
+                Instant.ofEpochMilli(iat.getTime()),
+                Instant.ofEpochMilli(exp.getTime()));
+
+            OAuth2AuthorizedClient oAuth2AuthorizedClient = new OAuth2AuthorizedClient(clientRegistration,
+                authenticationToken.getName(), oAuth2AccessToken);
+
+            request.setAttribute(oboAuthorizedClientAttributeName, (T) oAuth2AuthorizedClient);
+            return (T) oAuth2AuthorizedClient;
+        } catch (Throwable throwable) {
+            LOGGER.error("Failed to loadAuthorizedClient", throwable);
+        }
+        return null;
+    }
+
+    @Override
+    public void saveAuthorizedClient(OAuth2AuthorizedClient oAuth2AuthorizedClient, Authentication authentication,
+                                     HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) {
+    }
+
+    @Override
+    public void removeAuthorizedClient(String s, Authentication authentication,
+                                       HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) {
+    }
+
+    ConfidentialClientApplication getClientApplication(String registrationId) {
+        return confidentialClientApplicationMap.get(registrationId);
+    }
+
+    private ConfidentialClientApplication createApp(ClientRegistration clientRegistration) {
+
+        String authorizationUri = clientRegistration.getProviderDetails().getAuthorizationUri();
+
+        String authority = interceptAuthorizationUri(authorizationUri);
+
+        IClientSecret clientCredential = ClientCredentialFactory.createFromSecret(clientRegistration.getClientSecret());
+        try {
+            return ConfidentialClientApplication.builder(clientRegistration.getClientId(), clientCredential)
+                                                .authority(authority)
+                                                .build();
+        } catch (MalformedURLException e) {
+            LOGGER.error("Failed to create ConfidentialClientApplication", e);
+        }
+        return null;
```

Comment:
If the application happens to fail being created, it will cause NPE when load the authorized client. Maybe we should check whether the client is null in loadauthorizedclient method.